### PR TITLE
Updated recipe for v2.3.0

### PR DESCRIPTION
**Pull request by astrofrog**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by astrofrog** [comment]
@conda-forge-admin, please rerender

**Comment by astrofrog** [comment]
@conda-forge-admin, please rerender

**Comment by pllim** [comment]
```
conda_libmamba_solver.conda_build_exceptions.ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("c_linux-64")}
Encountered problems while solving:
  - nothing provides requested c_linux-64
```

**Comment by pllim** [comment]
Maybe that rings a bell to someone. I dunno what that error is about.

Do we need to pin like https://github.com/conda-forge/astropy-feedstock/blob/main/recipe/meta.yaml ?

**Comment by bsipocz** [comment]
See what #27 does for this, it seems to get through the dependency builds and errors out with pyregion-specific things.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,22 +21,28 @@ requirements:
...
-    - numpy        # [not (py==36 and win)]
-    - numpy         # [py==36 and win]
+    - setuptools-scm
     - cython
+    - extension-helpers
   run:
     - python
-    - pyparsing >=2.0.0
-    - astropy
-    - {{ pin_compatible('numpy') }}   # [not (py==36 and win)]
```
Comment (bsipocz) [inline recipe/meta.yaml:34]: I think the pin_conpatible part should stay until the 2.0 migration happens

**Comment by astrofrog** [comment]
Sounds good will investigate

**Comment by mwcraig** [comment]
@conda-forge-admin please rerender

**Pull request merged**